src/components/Form/Form.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useCallback, useState, useRef } from 'react';
+import FormControl from './components/FormControl';
+import FormSubmit from './components/FormSubmit';
+import { FormProvider, type FormContextType, type FormState } from './formContext';
+
+// Form 컴포넌트 props 타입
+interface FormProps {
+  initialValues?: Record<string, any>;
+  onSubmit: (values: Record<string, any>) => void;
+  children?: React.ReactNode;
+  className?: string;
+  style?: React.CSSProperties;
+}
```

Comment:
이렇게되면 제대로 된 타입 추론이 안돼서 다음과 같이 하면 좋아요

```suggestion
type FormProps<T extends object> = React.PropsWithChildren<{
  initialValues?: T;
  onSubmit: (values: T) => void;
  className?: string;
  style?: React.CSSProperties;
}>
```